services/resource-usage-tracker/src/simcore_service_resource_usage_tracker/resource_tracker_process_messages.py

```diff
@@ -0,0 +1,121 @@
+import logging
+from typing import cast
+
+from fastapi import FastAPI
+from fastapi.encoders import jsonable_encoder
+from models_library.rabbitmq_messages import (
+    RabbitResourceTrackingHeartbeatMessage,
+    RabbitResourceTrackingMessages,
+    RabbitResourceTrackingStartedMessage,
+    RabbitResourceTrackingStoppedMessage,
+    SimcorePlatformStatus,
+)
+from models_library.resource_tracker import ResourceTrackerServiceType, ServiceRunStatus
+from pydantic import parse_raw_as
+
+from .models.resource_tracker_service_run import (
+    ServiceRunCreate,
+    ServiceRunLastHeartbeatUpdate,
+    ServiceRunStoppedAtUpdate,
+)
+from .modules.db.repositories.resource_tracker import ResourceTrackerRepository
+
+_logger = logging.getLogger(__name__)
+
+
+async def process_message(
+    app: FastAPI, data: bytes  # pylint: disable=unused-argument
+) -> bool:
+    rabbit_message = parse_raw_as(RabbitResourceTrackingMessages, data)
+    rabbit_message_type = type(rabbit_message)
+
+    resource_tacker_repo: ResourceTrackerRepository = ResourceTrackerRepository(
+        db_engine=app.state.engine
+    )
+
+    if rabbit_message_type == RabbitResourceTrackingStartedMessage:
+        await _process_start_event(
+            resource_tacker_repo,
+            cast(RabbitResourceTrackingStartedMessage, rabbit_message),
+        )
+    elif rabbit_message_type == RabbitResourceTrackingHeartbeatMessage:
+        await _process_heartbeat_event(
+            resource_tacker_repo,
+            cast(RabbitResourceTrackingHeartbeatMessage, rabbit_message),
+        )
+    elif rabbit_message_type == RabbitResourceTrackingStoppedMessage:
+        await _process_stop_event(
+            resource_tacker_repo,
+            cast(RabbitResourceTrackingStoppedMessage, rabbit_message),
+        )
+    else:
+        raise NotImplementedError
```

Comment:
We can make this go faster, especially since we have lots of events incoming that require processing.
If the types of messages are mapped to the handlers then you just fetch the handler form a dict and execute. No need for lots of if statements.

```python
MESSAGE_TYPE_TO_HANDLER: dict[BaseMessageClassGoesHere, Awaitable[Callable[[ResourceTrackerRepository], None]] = {
   RabbitResourceTrackingStartedMessage:_process_start_event, 
...}
```

Reply:
and you no longer raise a not implemente error

Reply:
done :+1: